allennlp/commands/elmo.py

```diff
@@ -268,34 +277,50 @@ def embed_file(self,
             The embeddings to output.  Must be one of "all", "top", or "average".
         batch_size : ``int``, optional, (default = 64)
             The number of sentences to process in ELMo at one time.
+        use_sentence_keys : ``bool``, optional, (default = False).
+            Whether or not to use full sentences as keys. By default,
+            the line numbers of the input file are used as ids, which is more robust.
         """
 
         assert output_format in ["all", "top", "average"]
 
         # Tokenizes the sentences.
-        sentences = [line.strip() for line in input_file if line.strip()]
+        sentences = [line.strip() for line in input_file]
+
+        blank_lines = [i for (i, line) in enumerate(sentences) if line == ""]
+        if blank_lines:
+            raise ConfigurationError(f"Your input file contains empty lines at indexes "
+                                     f"{blank_lines}. Please remove them.")
         split_sentences = [sentence.split() for sentence in sentences]
-        # Uses the sentence as the key.
-        embedded_sentences = zip(sentences, self.embed_sentences(split_sentences, batch_size))
+        # Uses the sentence index as the key.
+
+        if use_sentence_keys:
+            logger.warning("Using sentences as keys can fail if sentences "
+                           "contain backslashes or colons. Use with caution.")
+            embedded_sentences = zip(sentences, self.embed_sentences(split_sentences, batch_size))
+        else:
+            embedded_sentences = enumerate(self.embed_sentences(split_sentences, batch_size))
```

Comment:
I would probably do the `str` here, so that you don't have to do it multiple times later (and so that embedded sentences has a nice type)

```
embedded_sentences = ((str(i), self.embed_sentences(sentence, batch_size)) for sentence in split_sentences)
```